### DISCUSSION TITLE
Default parameters are not the same when using the transcribe function in Python vs. using the Whisper commandline tool

### DISCUSSION THREAD

**Discussion by bmilde** [openai/whisper#591] [Q&A]
Using something like:

    import whisper
    model = whisper.load_model("large")
    result = model.transcribe("audio.mp3", language='de')

Gives different results compared to './whisper --language de --model large audio.mp3' for the same input. In particular, the transcription using the transcribe functions seems to fail sometimes with strange reptitions of the first segment (see below), while its completely fine with the command line. Looking through https://raw.githubusercontent.com/openai/whisper/main/whisper/transcribe.py, some default values only get passed to transcribe() through all of the argparse arguments with **args.

If I explicity set all parameters that are also passed as default values through **args, I can correctly transcribe my problematic input file:

     result = model.transcribe("audio.mp3", language=language, task='transcribe', temperature=(0.0, 0.2, 0.4, 0.6, 0.8, 1.0), best_of=5, beam_size=5, suppress_tokens="-1", condition_on_previous_text=True, fp16=True, compression_ratio_threshold=2.4, logprob_threshold=-1., no_speech_threshold=0.6)

It also takes about 2x longer to transcribe using the same hardware with the above parameters. Maybe its best_of=5 or beam_size=5 that only gets set through the argparse defaults?

Anyway, it would make sense if all of the default parameters are the same for the commandline version and the Python transcribe() function.

[degenerate transciption mentioned above:]

00:00.000 --> 00:20.000
Rap-Superstar

00:20.000 --> 00:42.000
Rap-Superstar

00:42.000 --> 01:05.000
Rap-Superstar

01:05.000 --> 01:25.000
Rap-Superstar

01:25.000 --> 01:45.000
Rap-Superstar

01:45.000 --> 02:05.000
Rap-Superstar

02:05.000 --> 02:25.000
Rap-Superstar

02:25.000 --> 02:45.000
Rap-Superstar

02:45.000 --> 03:05.000
Rap-Superstar

... + 100x more of the above.

**Comment by jianfch** [comment]
See #177 and #192.

**Comment by FurkanGozukara** [comment]
@bmilde does this generate exactly same output as command line interface?

 result = model.transcribe("audio.mp3", language=language, task='transcribe', temperature=(0.0, 0.2, 0.4, 0.6, 0.8, 1.0), best_of=5, beam_size=5, suppress_tokens="-1", condition_on_previous_text=True, fp16=True, compression_ratio_threshold=2.4, logprob_threshold=-1., no_speech_threshold=0.6)

**Reply by bmilde** [reply]
@FurkanGozukara yes, at least in my limited testing on some German inputs it seems to match the command line outputs with the above code now. Note that it doesn't completely eliminate the repetition bug as it then just happens on other files, overall it seems to happen less often though. But at least I could reproduce it with the command line as well.
